opwen_email_client/webapp/forms/register.py

```diff
@@ -0,0 +1,87 @@
+from os import getuid
+from pathlib import Path
+from pwd import getpwuid
+from shutil import chown
+
+from flask_wtf import FlaskForm
+from opwen_email_client.webapp.actions import ClientRegister
+from opwen_email_client.webapp.actions import RestartApp
+from opwen_email_client.webapp.config import AppConfig
+from opwen_email_client.webapp.config import root_domain
+from wtforms import StringField
+from wtforms import SubmitField
+
+
+class RegisterForm(FlaskForm):
+
+    client_name = StringField()
+    github_username = StringField()
+    github_token = StringField()
+    submit = SubmitField()
+
+    def register(self):
+        registration_details = self._setup_client()
+        opwen_settings = self._fetch_settings()
+        path = self._setup_path()
+        self._write_settings_to_file(path, opwen_settings, registration_details)
+        self._restart()
+
+    def _setup_client(self):
+        name = self.client_name.data.strip()
+        username = self.github_username.data.strip()
+        token = self.github_token.data.strip()
+        register = ClientRegister(name, username, token, None)
+        client_details = register()
+        return client_details
+
+    def _fetch_settings(self):
+        if AppConfig.RESTART_PATHS:
+            restart_paths_list = []
+            for key, value in AppConfig.RESTART_PATHS.items():
+                restart_paths_list.append('{}={}'.format(key, value))
+            restart_path = ','.join(restart_paths_list)
+        else:
+            restart_path = ''
+
+        return {
+            'OPWEN_APP_ROOT': AppConfig.APP_ROOT,
+            'OPWEN_STATE_DIRECTORY': AppConfig.STATE_BASEDIR,
+            'OPWEN_SESSION_KEY': AppConfig.SECRET_KEY,
+            'OPWEN_MAX_UPLOAD_SIZE_MB': AppConfig.MAX_UPLOAD_SIZE_MB,
+            'OPWEN_SIM_TYPE': AppConfig.SIM_TYPE,
+            'OPWEN_EMAIL_SERVER_HOSTNAME': AppConfig.EMAIL_SERVER_HOSTNAME,
+            'OPWEN_CLIENT_NAME': self.client_name.data.strip(),
+            'OPWEN_ROOT_DOMAIN': root_domain,
+            'OPWEN_RESTART_PATH': restart_path,
+        }
+
+    def _setup_path(self):
+        user = getpwuid(getuid()).pw_name
+        home = Path('/') / 'home' / user
+        path = Path('lokole/state/settings.env')
```

Comment:
I would suggest for the state location to respect the `LOKOLE_STATE_DIRECTORY` environment variable to be consistent with the [install.py code](https://github.com/ascoderu/lokole/blob/57bc073c632e0cfd526f544f54b1f0792a205e52/install.py#L883-L885):

```suggestion
        path = Path(getenv('LOKOLE_STATE_DIRECTORY', 'lokole/state')) / 'settings.env'
```